components/org.wso2.carbon.identity.remotefetch.core/src/main/java/org/wso2/carbon/identity/remotefetch/core/impl/handlers/action/WebHookHandler.java

```diff
@@ -0,0 +1,99 @@
+import static org.wso2.carbon.identity.remotefetch.common.RemoteFetchConstants.ID_UI_FIELD_URI;
+
+
+/**
+ * This class is used to handle web hook.{@see https://developer.github.com/webhooks/}
+ *
+ */
+public class WebHookHandler {
+
+    private String url;
+    private String branch;
+    private List<String> modifiedFileNames;
+    private RemoteFetchTaskExecutor remoteFetchTaskExecutor;
+    private RemoteFetchConfigurationDAO remoteFetchConfigurationDAO;
+
+    /**
+     * Constructor to create web hook handler.
+     * @param url Remote repository clone url.
+     * @param branch Remote repository branch.
+     * @param modifiedFileNames Modified file name list.
+     * @param remoteFetchTaskExecutor Task Executor to schedule immediate task for valid web hooks.
+     */
+    public WebHookHandler(String url, String branch, List<String> modifiedFileNames,
+                          RemoteFetchTaskExecutor remoteFetchTaskExecutor) {
+        this.url = url;
+        this.branch = branch;
+        this.modifiedFileNames = modifiedFileNames;
+        this.remoteFetchTaskExecutor = remoteFetchTaskExecutor;
+        remoteFetchConfigurationDAO = new RemoteFetchConfigurationDAOImpl();
+    }
+
+    /**
+     * This method is used to check if web hook matches with any remote fetch configuration for particular tenant.
+     * If so schedule a immediate task for matched remote fetch configuration.
+     * @throws RemoteFetchCoreException
+     */
+    public void handleWebHook() throws RemoteFetchCoreException {
+
+        int tenantId = IdentityTenantUtil.getTenantId(CarbonContext.getThreadLocalCarbonContext().getTenantDomain());
+        List<RemoteFetchConfiguration> remoteFetchConfigurations =
+                this.remoteFetchConfigurationDAO.getWebHookRemoteFetchConfigurationsByTenant(tenantId);
+        for (RemoteFetchConfiguration remoteFetchConfiguration: remoteFetchConfigurations) {
+
+            if (isWebHookMatches(remoteFetchConfiguration)) {
+                remoteFetchTaskExecutor.startImmediateTaskExecution(remoteFetchConfiguration);
+            }
+        }
+    }
+
+    /**
+     * Check whether given web hook params are matching with remote fetch configuration.
+     * @param remoteFetchConfiguration
+     * @return
+     */
+    private boolean isWebHookMatches(RemoteFetchConfiguration remoteFetchConfiguration) {
+
+        Map<String, String> repositoryManagerAttributes = remoteFetchConfiguration.getRepositoryManagerAttributes();
+
+        return (StringUtils.equalsIgnoreCase(url, repositoryManagerAttributes.get(ID_UI_FIELD_URI))) &&
+                (StringUtils.equalsIgnoreCase(branch, repositoryManagerAttributes.get(ID_UI_FIELD_BRANCH))) &&
+                modifiedFileNames.stream().anyMatch(modifiedFileName ->
+                        (modifiedFileName.startsWith(repositoryManagerAttributes.get(ID_UI_FIELD_DIRECTORY))));
+    }
+
```

Comment:
Unnecessary new line.

Reply:
Thank you, This comment and your previous comments were fixed in
https://github.com/wso2-extensions/identity-fetch-remote/pull/40
https://github.com/wso2-extensions/identity-fetch-remote/pull/41
https://github.com/wso2-extensions/identity-fetch-remote/pull/42
https://github.com/wso2-extensions/identity-fetch-remote/pull/43
https://github.com/wso2-extensions/identity-fetch-remote/pull/44
https://github.com/wso2-extensions/identity-fetch-remote/pull/45.
Hence I resolve this conversation.